src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java

```diff
@@ -384,12 +430,49 @@ public TarArchiveEntry(final File file, final String fileName) {
         } else {
             this.mode = DEFAULT_FILE_MODE;
             this.linkFlag = LF_NORMAL;
-            this.size = file.length();
+            try {
+                this.size = Files.size(file);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Not helpful.

Reply:
That is exactly where I wan't the feedback how to best handle it for backward compatibility 😉 See the second part in the description

Reply:
Let's not dump stacks to the console, very messy when used in a larger software stack.

Reply:
Yes, I know. This will **not** be in the final version of this PR!